File: app/core/entity/po/WorldEntity.py
from datetime import datetime

from pony.orm import PrimaryKey, Required, Set

from core.mapper.config.DatabaseConfig import db

# 世界观信息
class WorldEntity(db.Entity):
    world_id = PrimaryKey(int, auto=True)
    world_name = Required(str)
    world_desc = Required(str)
    create_time = Required(datetime)

    # 反向引用
    world_detail = Set('WorldDetailEntity')

    # 多对多，关联小说
    novel = Set('NovelEntity')


# 对应详细世界观信息
class WorldDetailEntity(db.Entity):
    world_detail_name = Required(str)
    world_detail_desc = Required(str)
    world = Required(WorldEntity)

File: front/src/entity/WorldEntity.ts
export interface createWorldDto {
    world_name: string;
    world_desc: string;
    create_time?: Date;
}

export interface createWorldDetailDto {
    world_detail_name: string;
    world_detail_desc: string;
    world: number;
}

export interface worldEntity {
    world_id: number;
    world_name: string;
    world_desc: string;
    create_time: Date;
}

export interface worldDetailEntity {
    id: number;
    world_detail_name: string;
    world_detail_desc: string;
    create_time: Date;
}

export interface allWorldDetailDto extends worldEntity {
    world_details: worldDetailEntity[];
}